gtests/test_nopython_serializable.cpp

```diff
@@ -0,0 +1,166 @@
+#include <gtest/gtest.h>
+#include <thread>
+
+#ifdef Py_PYTHON_H
+#error "Python.h should not be included."
+#endif
+
+#include <modmesh/serialization/SerializableItem.hpp>
+namespace modmesh
+{
+
+namespace detail
+{
+
+struct Address : SerializableItem
+{
+    std::string country;
+    std::string city;
+    std::vector<std::string> phone_numbers;
+    std::vector<int> zip_codes;
+
+    MM_DECL_SERIALIZABLE(
+        register_member("country", country);
+        register_member("city", city);
+        register_member("phone_numbers", phone_numbers);
+        register_member("zip_codes", zip_codes);)
+}; // end struct Address
+
+struct Pet : SerializableItem
+{
+    std::string name;
+    int age;
+    bool is_dog;
+    bool is_cat;
+
+    MM_DECL_SERIALIZABLE(
+        register_member("name", name);
+        register_member("age", age);
+        register_member("is_dog", is_dog);
+        register_member("is_cat", is_cat);)
+}; // end struct Pet
+
+struct Person : SerializableItem
+{
+    std::string name;
+    int age;
+    bool is_student;
+    Address address;
+    std::vector<Pet> pets;
+
+    MM_DECL_SERIALIZABLE(
+        register_member("name", name);
+        register_member("age", age);
+        register_member("is_student", is_student);
+        register_member("address", address);
+        register_member("pets", pets);)
+}; // end struct Person
+
+Pet create_dog()
+{
+    Pet pet;
+    pet.name = "Fluffy";
+    pet.age = 3;
+    pet.is_dog = true;
+    pet.is_cat = false;
+    return pet;
+}
+
+Pet create_cat()
+{
+    Pet pet;
+    pet.name = "Whiskers";
+    pet.age = 8;
+    pet.is_dog = false;
+    pet.is_cat = true;
+    return pet;
+}
+
+Address create_address()
+{
+    Address address;
+    address.country = "USA";
+    address.city = "New York";
+    address.phone_numbers = {"123-456-7890", "098-765-4321"};
+    address.zip_codes = {10001, 10002};
+    return address;
+}
+
+struct SecretItem : SerializableItem
```

Comment:
It's not intuitive to use "secret" to refer to "private", but it's not an issue either.